pkg/api/model/user.go

```diff
@@ -19,6 +19,7 @@ type User struct {
 	StandardAttributes map[string]interface{} `json:"standard_attributes,omitempty"`
 	CustomAttributes   map[string]interface{} `json:"custom_attributes,omitempty"`
 	Web3               *UserWeb3Info          `json:"x_web3,omitempty"`
+	LDAP               *UserLDAPInfo          `json:"x_ldap,omitempty"`
```

Comment:
Do not add this. This struct is part of the public API and we do not want to commit to expose this information.

I suggest the following
- Add a JSON ignored field `EndUserAccountID string `json:"-"`.
- This means you have to remove the original `EndUserAccountID()` method, as the name clashes.
- Implement Identity.Info.DisplayID for LDAP.
- In `newUserModel`, you construct `EndUserAccountID` with the information you have, that is `standardAttributes`, `identities`, and `web3Info`.

```
var endUserAccountID string

var ldapDisplayID string
for _, iden := range identities {
	if iden.Type == model.IdentityTypeLDAP {
		ldapDisplayID = iden.DisplayID()
		break
	}
}

if s, ok := u.StandardAttributes[string(ClaimEmail)].(string); ok && s != "" {
	endUserAccountID = s
} else if s, ok := u.StandardAttributes[string(ClaimPreferredUsername)].(string); ok && s != "" {
	endUserAccountID = s
} else if s, ok := u.StandardAttributes[string(ClaimPhoneNumber)].(string); ok && s != "" {
	endUserAccountID = s
} else if ldapDisplayID != "" {
	endUserAccountID = ldapDisplayID
} else if web3Info != nil && len(web3Info.Accounts) > 0 {
	first := web3Info.Accounts[0]
	endUserAccountID = first.EndUserAccountID()
}

```